drivers/adc/adc_tla2021.c

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright (c) 2023 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <stdint.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+
+/*
+ * This requires to be included _after_  `#define ADC_CONTEXT_USES_KERNEL_TIMER`
+ */
+#include "adc_context.h"
+
+#define DT_DRV_COMPAT ti_tla2021
+
+LOG_MODULE_REGISTER(tla2021, CONFIG_ADC_LOG_LEVEL);
+
+/*
+ * Conversion Data Register (RP = 00h) [reset = 0000h]
+ */
+#define REG_DATA     0x00
+#define REG_DATA_pos 4
+
+/*
+ * Configuration Register (RP = 01h) [reset = 8583h]
+ */
+#define REG_CONFIG          0x01
+#define REG_CONFIG_DEFAULT  0x8583
+#define REG_CONFIG_DR_pos   5
+#define REG_CONFIG_MODE_pos 8
+#define REG_CONFIG_PGA_pos  9  /* TLA2022 and TLA2024 Only */
+#define REG_CONFIG_MUX_pos  12 /* TLA2024 Only */
+#define REG_CONFIG_OS_pos   15
+#define REG_CONFIG_OS_msk   (BIT_MASK(1) << REG_CONFIG_OS_pos)
+
+#define ACQUISITION_THREAD_STACK_SIZE 512
+#define ACQUISITION_THREAD_PRIORITY   80
```

Comment:
I would suggest adding these to Kconfig, but also I defer to @anangl in case there is already some established convention here.